### PR TITLE
Basic templating for blogs

### PR DESCRIPTION
**Pull request by madeleinefrancis**
Still having trouble with sizing of the mat-dialog used for blogDialog components when in mobile view... not sure where it's inheriting its small size from. I'll expand the post class to include photos once I know how that will work with you guys.

### CODE REVIEW

**Comment by omzmarlon** [comment]
Good Work! I'll take a look ASAP

File: web/src/redcross/blogModule/components/BlogComponent.ts [hunk 1]
```diff
@@ -1,10 +1,45 @@
...
+        public getData(): void {
+        this.posts = this.dds.getTestPosts();
+        this.currentUser = this.dds.getTestAuthor();
+    }
+
+        private ngOnInit(): void {
+        this.getData();
+    }
+
+        private addPost(): void {
```
Comment (omzmarlon) [inline web/src/redcross/blogModule/components/BlogComponent.ts]: this line not properly formatted

File: web/src/redcross/blogModule/components/BlogComponent.ts [hunk 2]
```diff
@@ -1,10 +1,45 @@
...
+
+    constructor(public dds: DiscussionDataService, public md: MatDialog) {}
+
+        // loads test data
+        public getData(): void {
+        this.posts = this.dds.getTestPosts();
+        this.currentUser = this.dds.getTestAuthor();
+    }
+
+        private ngOnInit(): void {
```
Comment (omzmarlon) [inline web/src/redcross/blogModule/components/BlogComponent.ts]: this line not properly formatted

File: web/src/redcross/blogModule/components/BlogDataService.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import {Injectable} from "@angular/core";
+
+import {Post} from "./Post";
+import {Author} from "./Author";
+
+@Injectable()
+export class DiscussionDataService {
```
Comment (omzmarlon) [inline web/src/redcross/blogModule/components/BlogDataService.ts]: make your class name match file name. And also why is it called discussion?

File: web/src/redcross/blogModule/components/BlogComponent.ts [hunk 4]
```diff
@@ -1,10 +1,45 @@
-import {Component} from "@angular/core";
+import {Component, OnInit, Inject} from "@angular/core";
+import {BlogPostComponent} from "./BlogPostComponent";
+import {BlogDialogComponent} from "./BlogDialogComponent";
+import {MatDialog, MatDialogRef, MAT_DIALOG_DATA} from "@angular/material";
```
Comment (omzmarlon) [inline web/src/redcross/blogModule/components/BlogComponent.ts]: remove unused imports

File: web/src/redcross/blogModule/components/BlogComponent.ts [hunk 5]
```diff
@@ -1,10 +1,45 @@
...
-    styleUrls: ["./Blog.less"]
+    styleUrls: ["./Blog.less"],
+    providers: [DiscussionDataService, MatDialog]
 })
 export class BlogComponent {
 
+    private posts: Post[];
+    private currentUser: Author;
+
+    constructor(public dds: DiscussionDataService, public md: MatDialog) {}
```
Comment (omzmarlon) [inline web/src/redcross/blogModule/components/BlogComponent.ts]: these should be private. If you do intend to let other classes to use them, always make them private! Check other classes constructor too!

**Review by omzmarlon — CHANGES_REQUESTED** [review]
@madeleinefrancis 
Just took a look. You've done great work! : )
A few comments:
1. I have a cannot read title of undefined in my browser console, when I navigate to http://localhost:8000/#/blogs/postTest
2. Great that you've spot a need for model classes like Post.ts and Author.ts. 
But I would suggest creating a new folder inside blogModule called "models" and then move these to there
3. The MatDialogModule you imported in BlogModule.ts, since many different modules will need MatDialog, so I would suggest move it there, and any other modules from Angular Material. This will probably cause conflict with other people's code when you merge to master. But we can deal with that later
4. I'm not sure why you added child routes to BlogRoutingModule?
5. For the add button, you can use MatIcon from Angular material, sth like:
`<button md-fab><md-icon>add</md-icon></button>`
The icon should work once you use them, since I think've add a link to get icons globally in index.html. If your code can't find the icon, let me know
6. Same for services. Create a folder called "services" and move the BlogDataService there
7. rename BlogComponent to BlogContainerComponent
8. for all of your less files, use nesting to better reflect structures in html. eg: in Blog.less, move everything inside .blog-outer
9. other comments are with the code

I might have just a few other things to say(about photos and structuring container and children components), but fix these first we can discuss the rest after that!

**Comment by omzmarlon** [comment]
Not nesting styles in BlogDialog.less
You add button's icon is not working. And also has a really weird box shadow. 
When I try to add a new blog, an error is thrown. 
See below picture:
aaa

**Comment by madeleinefrancis** [comment]
So not sure if using <i class="material-icons"> to get the icon working is okay but having trouble getting it to show with <md-icon>

**Comment by madeleinefrancis** [comment]
Not sure if its desirable but I was having trouble getting the icon to show with md-icon so I ended up using class="material-icons" in an "i" element.

**Comment by madeleinefrancis** [comment]
Also just added a background color to the icon button because it becomes barely visible in mobile view but I guess we will probably change the styles later anyways.

File: web/src/redcross/blogModule/components/Blog.ng.html [hunk 6]
```diff
@@ -1,7 +1,8 @@
 <div class="blog-outter">
-    blog
     <blogPost *ngFor="let p of posts" [post]="p">
     </blogPost>
-    <button class="add-post-button" md-fab (click)="addPost()"><md-icon>add</md-icon></button>
+    <button md-fab class="add-post-button" (click)="addPost()">
```
Comment (omzmarlon) [inline web/src/redcross/blogModule/components/Blog.ng.html]: You can add button by using the mat-icon-button with mat-icon nested inside. Something like:
`<button mat-icon-button>
<mat-icon>add</mat-icon>
</button>`
Using mat-icon, Angular Material should automatically load.
Checkout Angular Material's documentation on buttons and mat-icon.

**Review by omzmarlon — CHANGES_REQUESTED** [review]
There's just one more small problem with the create new blog post form. When I click cancel, a new blog post was still created and added to the list

File: web/src/redcross/blogModule/components/Blog.ng.html (hunk 6 above)
Comment (omzmarlon) [reply web/src/redcross/blogModule/components/Blog.ng.html]: Or, if you really can't figure it out, checkout  Justin's code. He basically put a '+' sign there. 
This is not the best approach, but we can change that later.
His code is already merged to master.

**Review by omzmarlon — APPROVED** [review]

**Comment by omzmarlon** [comment]
Good job!

**Pull request merged**